fix: Upload remove margin using grid display

CooperHash opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is confirmed.
Your pull requests will be merged after one of the collaborators approve.
Thank you!
-->

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [X] Other (about what?)

### 🔗 Related issue link

- close #47992

<!--
1. Put the related issue or discussion links here.
2. close #xxxx or fix #xxxx for instance.
-->

### 💡 Background and solution

改动前

![image](https://github.com/ant-design/ant-design/assets/73218815/8fcb759f-8d08-4780-a9e4-8c5c4a286201)

改动后

![image](https://github.com/ant-design/ant-design/assets/73218815/0730ed1b-fd02-4e30-bba2-622f0e422452)



<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list the final API implementation and usage sample if that is a new feature.
-->

### 📝 Changelog

<!--
Describe changes from the user side, and list all potential break changes or other risks.
--->

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |    --       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

CooperHash committed:
fix: Upload remove margin using grid display

li-jia-nan commented on components/upload/style/picture.ts:
```diff
@@ -138,19 +136,26 @@ const genPictureCardStyle: GenerateStyle<UploadToken> = (token) => {
 
       // list
       [`${listCls}${listCls}-picture-card, ${listCls}${listCls}-picture-circle`]: {
+        display: 'grid',
```
为啥不用 flex ？

CooperHash replied:
没做太多思考，这里用flex会更好吗

li-jia-nan replied:
我是看到 #47992 这个 issue 中讨论的是 flex，grid 布局在 antd 里面用的比较少

CooperHash committed:
change grid to flex

li-jia-nan commented on components/upload/style/picture.ts:
```diff
@@ -138,19 +136,25 @@ const genPictureCardStyle: GenerateStyle<UploadToken> = (token) => {
       [`${listCls}${listCls}-picture-card, ${listCls}${listCls}-picture-circle`]: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: `${token.marginXS} ${token.marginXS}`,
```
```suggestion
        gap: token.marginXS,
```

CooperHash committed:
fix

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: Cooper. <73218815+CooperHash@users.noreply.github.com>

li-jia-nan commented:
感觉这个得放在 feature 分支，我不确定，大佬们啥意见？@zombieJ @MadCcc

li-jia-nan committed:
chore: rename variables `merge` => `merged` (#48048)

afc163 commented:
+1 feature 分支

CooperHash commented:
那我是关掉这个分支，再重写提交一次到feature吗

li-jia-nan commented:
可以直接改，上面有编辑按钮

CooperHash committed:
docs: Add FAQ to popconfirm (#48055)

li-jia-nan committed:
fix: adjust the order of QRCode style prop (#48053)

* fix: adjust the order of style prop

* test: update snap

* test: add test case

MadCcc committed:
docs: fix anchor flick when has popover (#48064)

crazyair committed:
docs: 修复 menu openKeys 计算问题 (#47859)

* fix: 修复 menu openKeys 计算问题

* feat: keys

* feat: 优化代码,避免 eslint

* feat: 优化代码,避免 eslint

* feat: 3级增加为3条数据,为了换行,便于代码查看

* feat: 删除受控 selectedKeys

* feat: 优化代码

* feat: snap

* feat: 优化代码去掉lint

* feat: 代码优化

* feat: 删除 as 因为 onOpenChange 不是 React.Key 了

* feat: 优化代码

* feat: 添加删移除所有子集逻辑

Meetacoo committed:
demo: Update demo with Space.Compact wrap (#48067)

* demo: App demo update

* test: Update test snapshots

chenshuai2144 committed:
docs: add 5.15.4 chaneglog (#48057)

* docs: add 5.15.4 chaneglog

* fix typo

* Update CHANGELOG.zh-CN.md

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* Update CHANGELOG.en-US.md

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* fix

* fix

* fix lint

* Update CHANGELOG.en-US.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* fix

* fix

---------

Signed-off-by: 陈帅 <qixian.cs@outlook.com>
Co-authored-by: lijianan <574980606@qq.com>
Co-authored-by: afc163 <afc163@gmail.com>

kiner-tang committed:
feat: optimize code (#48072)

MadCcc committed:
fix: color picker controlled value (#47816)

* fix: fix control value not show in dom

* chore: update title

* add test case

* change test title

* fix: cleared color should change after controlled value changed

* chore: code clean

* test: change test

* comment

* fix: should respect empty string

* test: add test case

* chore: update demo

---------

Co-authored-by: tanghui <yoyo837@hotmail.com>
Co-authored-by: afc163 <afc163@gmail.com>
Co-authored-by: lijianan <574980606@qq.com>

dependabot[bot] committed:
chore(deps-dev): update sharp requirement from ^0.33.2 to ^0.33.3 (#48078)

Updates the requirements on [sharp](https://github.com/lovell/sharp) to permit the latest version.
- [Release notes](https://github.com/lovell/sharp/releases)
- [Changelog](https://github.com/lovell/sharp/blob/main/docs/changelog.md)
- [Commits](https://github.com/lovell/sharp/compare/v0.33.2...v0.33.3)

---
updated-dependencies:
- dependency-name: sharp
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump the dev-dependencies group with 4 updates (#48077)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  dependency-group: dev-dependencies
- dependency-name: antd-style
  dependency-type: direct:development
  dependency-group: dev-dependencies
- dependency-name: react-countup
  dependency-type: direct:development
  dependency-group: dev-dependencies
- dependency-name: stylelint
  dependency-type: direct:development
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

yoshinyan committed:
docs: typo in order-column.md (#48079)

Signed-off-by: yoshinyan <hai_hai_hai_1229@yahoo.co.jp>

crazyair committed:
feat: deprecated-codesandbox (#48081)

* feat: deprecated-codesandbox

* feat: debug show csb

MadCcc commented on components/upload/style/picture.ts:
```diff
@@ -138,19 +136,25 @@ const genPictureCardStyle: GenerateStyle<UploadToken> = (token) => {
       [`${listCls}${listCls}-picture-card, ${listCls}${listCls}-picture-circle`]: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: token.marginXS,
```
gap 在 chromium 84 版本以上才支持，需要用 @supports 做下降级

TheKonka committed:
site: fix the aside nav may cover the version dropdown box in so… (#48082)

* fix(site): fix the aside nav may cover the version dropdown box in some screen resolutions

* fix(review): fix code  review

CooperHash committed:
Merge branch 'feature' into remove-margin

CooperHash committed:
Merge branch 'ant-design:master' into remove-margin

CooperHash commented:
git被我搞炸了...我重新开一个branch吧...

CooperHash commented:
#48091
